### PR TITLE
Add new PreRequisites Checker Gramplet[gramps50]

### PR DESCRIPTION
**Pull request by sam-m888**
A diagnostic tool to help work out if users have the [prerequisites](https://github.com/gramps-project/gramps/blob/master/README.md) installed as well as the latest version of Gramps

Also helps on OS's like windows and mac where it's difficult for users to use the command line to get information on gramps via ( gramps -v )!

- Retrieved and updated for Gramps 5

- Added simple Gramps version check

- Displays additional requirements for development & translation.

- Expanded to check all known third-party addons prerequisites

- Tested with Gramps versions: 3.3 to master (Only checked that it loads and displays on older versions)

- Correct min versions for GTK & pygobject (prculley)

- Add check for editexifmetadata addon (prculley)

- Add check for PIL to latexdoc type addons:genealogytree(prculley)

- Add check for PhotoTagging & lxml gramplets (PQYPLZXHGF)

- Add MongoDB (Dale)

context-menu-option

50prerequisites checker gramplet01
50prerequisites checker gramplet02

### CODE REVIEW

**Comment by adamvazquez82** [comment]
@sam-m888 Surprised something like this doesn't already exist for Gramps!  Thank you 👍 

I understand it's a WIP so some feedback is that it does not find that Google Earth **Pro** is installed on my win 10 system, but when I installed the standard version it finds it. 

Also separate to this but related; the GoogleMapKML addon does not work on 4.2.6 (creates an empty xml place file) and on the newest 5.0 alpha3 from @bpisoj I get

> 2018-02-14 15:21:27.187: WARNING: _manager.py: line 295: Plugin error (from 'GoogleEarthWriteKML'): cannot import name 'conv_to_unicode'

**Comment by prculley** [comment]
I think Gtk needs a min 3.10 version...  And, despite what our readme says, pygobject needs at least the version of gtk or higher or we will get crashes.

**Comment by sam-m888** [comment]
@prculley   Good catch I was testing for gtk 3.1.0 and not 3.10! Thanks for the feedback .

**Comment by sam-m888** [comment]
> I think Gtk needs a min 3.10 version... And, despite what our readme says, pygobject needs at least the version of installed gtk or higher or we will get crashes.

@prculley Corrected in latest push. Thanks

**Comment by sam-m888** [comment]
> And PIL (Pillow) is also used by the main Gramps (narrativeweb and other reports for image cropping) as well as editexifmetadata and the new latexdoc type addons.

@prculley Add check for PIL to editexifmetadata  and latexdoc type addons in latest push. Thanks

**Comment by PQYPLZXHGF** [comment]
Can you add PhotoTagging (has opencv & numpy) / lxml gramplet (has lxml)?

Very handy 🥇

**Comment by sam-m888** [comment]
> Can you add PhotoTagging (has opencv & numpy) / lxml gramplet (has lxml)?

@PQYPLZXHGF Added as requested.

**Comment by PQYPLZXHGF** [comment]
@sam-m888 Thanks 👍

**Comment by daleathan** [comment]
> Add MongoDB (Dale)

@sam-m888 That was quick, thanks.

File: PrerequisitesCheckerGramplet/PrerequisitesCheckerGramplet.py [hunk 1]
```diff
@@ -0,0 +1,1623 @@
...
+        * PILLOW
+
+        #TODO need to add to readme
+
+        https://github.com/gramps-project/addons-source/tree/maintenance/gramps50/GenealogyTree
+
+        https://github.com/gramps-project/gramps/blob/maintenance/gramps50/gramps/gen/plug/docgen/treedoc.py
+        '''
+        self.append_text("\n")
+        self.render_text("""<b>05. <a href="https://gramps-project.org/bugs/view.php?id=10223">Addon:GenealogyTrees</a> :</b> """)
```
Comment (adamvazquez82) [inline PrerequisitesCheckerGramplet/PrerequisitesCheckerGramplet.py]: https://gramps-project.org/wiki/index.php?title=Addon:GenealogyTree

**Comment by sam-m888** [comment]
*@adamvazquez82 *Url added

Thanks

**Comment by adamvazquez82** [comment]
@sam-m888 Thanks 😀 

Any word on when Gramps 5 will be released?

**Comment by prculley** [comment]
I'm sure we will eventually have more to add, but I think this is ready for a release 😀.

**Comment by daleathan** [comment]
> Any word on when Gramps 5 will be released?

@adamvazquez82 Looks like the only issue left on the [roadmap](https://gramps-project.org/bugs/roadmap_page.php?version_id=96) to be fixed is one raised by [yourself](https://gramps-project.org/bugs/view.php?id=10507) and the related PR https://github.com/gramps-project/gramps/pull/584 is yet to be accepted.  My quick test shows it works for me also 👍 Whether that means Gramps will be released soon after I don't know and only the developers will have to comment.

**Comment by adamvazquez82** [comment]
@daleathan Thanks 👍

**Comment by sam-m888** [comment]
> I'm sure we will eventually have more to add, but I think this is ready for a release

@prculley Ok haven't had much time lately to work on this addon, so I'll release it as is because of it's usefulness; but if before the release of Gramps 5 the this addon has any detrimental issues we can always unpublish the addon.

**Comment by adamvazquez82** [comment]
> @adamvazquez82 Looks like the only issue left on the roadmap to be fixed is one raised by yourself 

@daleathan Thanks, [it](https://github.com/gramps-project/gramps/pull/584) was accepted earlier today. Let hope that it will be released very soon 👍

File: PrerequisitesCheckerGramplet/PrerequisitesCheckerGramplet.py [hunk 2]
```diff
@@ -0,0 +1,1623 @@
...
+            #print("Failed")
+            messagefailed1 = " (Requires version "
+            messagefailed3= " or greater installed.)\n"
+
+            messagefailed= messagefailed1 + min_py_str + messagefailed3
+
+            result= check1 + py_str + messagefailed
+        else:
+            #print("Success")
+            messagesuccess1 = " (Successs version "
```
Comment (adamvazquez82) [inline PrerequisitesCheckerGramplet/PrerequisitesCheckerGramplet.py]: Successs  > Success

**Comment by adamvazquez82** [comment]
☹ Guess it'll be a while longer.

File: PrerequisitesCheckerGramplet/PrerequisitesCheckerGramplet.py (hunk 2 above)
Comment (sam-m888) [reply PrerequisitesCheckerGramplet/PrerequisitesCheckerGramplet.py]: Thanks fixed in https://github.com/gramps-project/addons-source/pull/144

**Pull request merged**